doc/vscode-config.md

```diff
@@ -0,0 +1,253 @@
+# Vscode必备插件
```

Comment:
Since developers use a wide variety of tools (CLion, Vim, Emacs, VS Code, etc.), IDE configuration is usually considered a personal preference. Adding a guide for one specific IDE might imply it's the 'official' or only supported way. Additionally, keeping this documentation up-to-date as VS Code extensions change can become a maintenance burden.
I feel this is out of scope for the core repository. Perhaps this content would be better suited for the project's Wiki or blog?